src/recipe_groups.cpp

```diff
@@ -93,12 +93,43 @@ std::map<recipe_id, translation> recipe_group::get_recipes_by_id( const std::str
     }
     const recipe_group_data &group = recipe_groups_data.obj( group_id( id ) );
     if( om_terrain_id != "ANY" ) {
+        std::string base_om_ter_id = om_terrain_id;
+        size_t om_ter_len = om_terrain_id.size();
+        if( om_ter_len > 7 ) {
+            if( om_terrain_id.substr( om_ter_len - 6, 6 ) == "_south" ) {
+                base_om_ter_id = om_terrain_id.substr( 0, om_ter_len - 6 );
+            } else if( om_terrain_id.substr( om_ter_len - 6, 6 ) == "_north" ) {
+                base_om_ter_id = om_terrain_id.substr( 0, om_ter_len - 6 );
+            }
+        }
+        if( om_ter_len > 6 ) {
```

Comment:
It's nice of you to split this check for length 7 and length 6 here, but why did you not keep this up in the checks below? (The below won't catch a terrain named "tt_east" is has not more than 7 characters).

As I've often seen this kind of code around the mapgen code, it should really be put into a separate function, instead of repeating it (partially wrong) over and over again.

Reply:
Functions added, thanks for the suggestion.

Reply:
Also, please don't add sarcastic comments in another commit instead of fixing a problem; you just caused an unnecessary merge conflict.